sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -988,6 +990,38 @@ def custom_metric(x1, x2):
     assert_array_almost_equal(dist1, dist2)
 
 
+def test_algo_auto_metrics():
+    X = rng.rand(12, 12)
+    Xcsr = csr_matrix(X)
+
+    # Metric which don't required any additional parameter
+    _metrics = ['mahalanobis', 'wminkowski', 'seuclidean']
+    for metric in VALID_METRICS['brute']:
+        if metric not in _metrics:
+            nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
+                                            metric=metric).fit(X)
+            nn.kneighbors(X)
+    for metric in VALID_METRICS_SPARSE['brute']:
+        if metric not in _metrics:
+            nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
+                                            metric=metric).fit(Xcsr)
+            if metric != "precomputed":
```

Comment:
That case is not implemented: it would need to be treated a bit differently to the precomputed dense array case; and it doesn't really make sense. An array containing distances which is mostly 0s is pretty useless for nearest neighbors. (Unless we're admitting negative distances, or having a different interpretation of 0s, both of which may be useful.)